scripts/typescript_checks.py

```diff
@@ -77,7 +77,8 @@ def compile_and_check_typescript(config_path):
     process = subprocess.Popen(cmd, stdout=subprocess.PIPE)
     error_messages = []
     for line in iter(process.stdout.readline, ''):
-        error_messages.append(line)
+        if not line.startswith('node_modules'):
+            error_messages.append(line)
```

Comment:
Hmm, not sure about this one, but it probably makes sense to skip `node_modules`, could you show the errors produced in `node_modules`?

Reply:
```
node_modules/@types/node/module.d.ts(51,14): error TS2300: Duplicate identifier 'mod'.

node_modules/@types/node/ts3.5/globals.global.d.ts(1,13): error TS2403: Subsequent variable declarations must have the same type.  Variable 'global' must be of type 'Global', but here has type 'Global & typeof globalThis'.

node_modules/@types/requirejs/index.d.ts(38,14): error TS2300: Duplicate identifier 'mod'.

node_modules/@types/requirejs/index.d.ts(422,13): error TS2403: Subsequent variable declarations must have the same type.  Variable 'require' must be of type 'NodeRequire', but here has type 'Require'.

node_modules/typescript/lib/lib.dom.d.ts(25,1): error TS6200: Definitions of the following identifiers conflict with those in another file: EventListenerOrEventListenerObject, ImportExportKind, TableKind, ValueType, ExportValue, Exports, ImportValue, ModuleImports, Imports, name, HeadersInit, BodyInit, RequestInfo, BlobPart, DOMHighResTimeStamp, CanvasImageSource, OffscreenRenderingContext, MessageEventSource, ImageBitmapSource, OnErrorEventHandler, TimerHandler, PerformanceEntryList, ReadableStreamReadResult, VibratePattern, AlgorithmIdentifier, HashAlgorithmIdentifier, BigInteger, NamedCurve, GLenum, GLboolean, GLbitfield, GLint, GLsizei, GLintptr, GLsizeiptr, GLuint, GLfloat, GLclampf, TexImageSource, Float32List, Int32List, GLint64, GLuint64, Uint32List, BufferSource, DOMTimeStamp, FormDataEntryValue, IDBValidKey, Transferable, BinaryType, CanvasDirection, CanvasFillRule, CanvasLineCap, CanvasLineJoin, CanvasTextAlign, CanvasTextBaseline, ClientTypes, ColorSpaceConversion, EndingType, IDBCursorDirection, IDBRequestReadyState, IDBTransactionMode, ImageOrientation, ImageSmoothingQuality, KeyFormat, KeyType, KeyUsage, NotificationDirection, NotificationPermission, OffscreenRenderingContextId, PermissionName, PermissionState, PremultiplyAlpha, PushEncryptionKeyName, PushPermissionState, ReferrerPolicy, RequestCache, RequestCredentials, RequestDestination, RequestMode, RequestRedirect, ResizeQuality, ResponseType, ServiceWorkerState, ServiceWorkerUpdateViaCache, VisibilityState, WebGLPowerPreference, WorkerType, XMLHttpRequestResponseType

node_modules/typescript/lib/lib.dom.d.ts(3713,5): error TS2687: All declarations of 'privateKey' must have identical modifiers.

node_modules/typescript/lib/lib.dom.d.ts(3714,5): error TS2687: All declarations of 'publicKey' must have identical modifiers.

node_modules/typescript/lib/lib.webworker.d.ts(25,1): error TS6200: Definitions of the following identifiers conflict with those in another file: EventListenerOrEventListenerObject, ImportExportKind, TableKind, ValueType, ExportValue, Exports, ImportValue, ModuleImports, Imports, name, HeadersInit, BodyInit, RequestInfo, BlobPart, DOMHighResTimeStamp, CanvasImageSource, OffscreenRenderingContext, MessageEventSource, ImageBitmapSource, OnErrorEventHandler, TimerHandler, PerformanceEntryList, ReadableStreamReadResult, VibratePattern, AlgorithmIdentifier, HashAlgorithmIdentifier, BigInteger, NamedCurve, GLenum, GLboolean, GLbitfield, GLint, GLsizei, GLintptr, GLsizeiptr, GLuint, GLfloat, GLclampf, TexImageSource, Float32List, Int32List, GLint64, GLuint64, Uint32List, BufferSource, DOMTimeStamp, FormDataEntryValue, IDBValidKey, Transferable, BinaryType, CanvasDirection, CanvasFillRule, CanvasLineCap, CanvasLineJoin, CanvasTextAlign, CanvasTextBaseline, ClientTypes, ColorSpaceConversion, EndingType, IDBCursorDirection, IDBRequestReadyState, IDBTransactionMode, ImageOrientation, ImageSmoothingQuality, KeyFormat, KeyType, KeyUsage, NotificationDirection, NotificationPermission, OffscreenRenderingContextId, PermissionName, PermissionState, PremultiplyAlpha, PushEncryptionKeyName, PushPermissionState, ReferrerPolicy, RequestCache, RequestCredentials, RequestDestination, RequestMode, RequestRedirect, ResizeQuality, ResponseType, ServiceWorkerState, ServiceWorkerUpdateViaCache, VisibilityState, WebGLPowerPreference, WorkerType, XMLHttpRequestResponseType

node_modules/typescript/lib/lib.webworker.d.ts(89,5): error TS2687: All declarations of 'privateKey' must have identical modifiers.

node_modules/typescript/lib/lib.webworker.d.ts(90,5): error TS2687: All declarations of 'publicKey' must have identical modifiers.

node_modules/typescript/lib/lib.webworker.d.ts(1155,13): error TS2403: Subsequent variable declarations must have the same type.  Variable 'DOMMatrixReadOnly' must be of type '{ new (init?: string | number[]): DOMMatrixReadOnly; prototype: DOMMatrixReadOnly; fromFloat32Array(array32: Float32Array): DOMMatrixReadOnly; fromFloat64Array(array64: Float64Array): DOMMatrixReadOnly; fromMatrix(other?: DOMMatrixInit): DOMMatrixReadOnly; toString(): string; }', but here has type '{ new (init?: string | number[]): DOMMatrixReadOnly; prototype: DOMMatrixReadOnly; fromFloat32Array(array32: Float32Array): DOMMatrixReadOnly; fromFloat64Array(array64: Float64Array): DOMMatrixReadOnly; fromMatrix(other?: DOMMatrixInit): DOMMatrixReadOnly; }'.

node_modules/typescript/lib/lib.webworker.d.ts(1252,5): error TS2375: Duplicate number index signature.

node_modules/typescript/lib/lib.webworker.d.ts(1561,5): error TS2375: Duplicate number index signature.

node_modules/typescript/lib/lib.webworker.d.ts(1635,13): error TS2403: Subsequent variable declarations must have the same type.  Variable 'FormData' must be of type '{ new (form?: HTMLFormElement): FormData; prototype: FormData; }', but here has type '{ new (): FormData; prototype: FormData; }'.

node_modules/typescript/lib/lib.webworker.d.ts(2192,14): error TS2717: Subsequent property declarations must have the same type.  Property 'canvas' must be of type 'HTMLCanvasElement | OffscreenCanvas', but here has type 'OffscreenCanvas'.

node_modules/typescript/lib/lib.webworker.d.ts(2380,13): error TS2403: Subsequent variable declarations must have the same type.  Variable 'Notification' must be of type '{ new (title: string, options?: NotificationOptions): Notification; prototype: Notification; readonly maxActions: number; readonly permission: NotificationPermission; requestPermission(deprecatedCallback?: NotificationPermissionCallback): Promise<...>; }', but here has type '{ new (title: string, options?: NotificationOptions): Notification; prototype: Notification; readonly maxActions: number; readonly permission: NotificationPermission; }'.

node_modules/typescript/lib/lib.webworker.d.ts(4778,14): error TS2717: Subsequent property declarations must have the same type.  Property 'canvas' must be of type 'HTMLCanvasElement | OffscreenCanvas', but here has type 'OffscreenCanvas'.

node_modules/typescript/lib/lib.webworker.d.ts(5887,13): error TS2403: Subsequent variable declarations must have the same type.  Variable 'onmessage' must be of type '(this: Window, ev: MessageEvent<any>) => any', but here has type '(this: DedicatedWorkerGlobalScope, ev: MessageEvent<any>) => any'.

node_modules/typescript/lib/lib.webworker.d.ts(5888,13): error TS2403: Subsequent variable declarations must have the same type.  Variable 'onmessageerror' must be of type '(this: Window, ev: MessageEvent<any>) => any', but here has type '(this: DedicatedWorkerGlobalScope, ev: MessageEvent<any>) => any'.

node_modules/typescript/lib/lib.webworker.d.ts(5905,13): error TS2403: Subsequent variable declarations must have the same type.  Variable 'location' must be of type 'Location', but here has type 'WorkerLocation'.

node_modules/typescript/lib/lib.webworker.d.ts(5906,13): error TS2403: Subsequent variable declarations must have the same type.  Variable 'navigator' must be of type 'Navigator', but here has type 'WorkerNavigator'.

node_modules/typescript/lib/lib.webworker.d.ts(5907,13): error TS2403: Subsequent variable declarations must have the same type.  Variable 'onerror' must be of type 'OnErrorEventHandlerNonNull', but here has type '(this: DedicatedWorkerGlobalScope, ev: ErrorEvent) => any'.

node_modules/typescript/lib/lib.webworker.d.ts(5908,13): error TS2403: Subsequent variable declarations must have the same type.  Variable 'onlanguagechange' must be of type '(this: Window, ev: Event) => any', but here has type '(this: DedicatedWorkerGlobalScope, ev: Event) => any'.

node_modules/typescript/lib/lib.webworker.d.ts(5909,13): error TS2403: Subsequent variable declarations must have the same type.  Variable 'onoffline' must be of type '(this: Window, ev: Event) => any', but here has type '(this: DedicatedWorkerGlobalScope, ev: Event) => any'.

node_modules/typescript/lib/lib.webworker.d.ts(5910,13): error TS2403: Subsequent variable declarations must have the same type.  Variable 'ononline' must be of type '(this: Window, ev: Event) => any', but here has type '(this: DedicatedWorkerGlobalScope, ev: Event) => any'.

node_modules/typescript/lib/lib.webworker.d.ts(5911,13): error TS2403: Subsequent variable declarations must have the same type.  Variable 'onrejectionhandled' must be of type '(this: Window, ev: PromiseRejectionEvent) => any', but here has type '(this: DedicatedWorkerGlobalScope, ev: PromiseRejectionEvent) => any'.

node_modules/typescript/lib/lib.webworker.d.ts(5912,13): error TS2403: Subsequent variable declarations must have the same type.  Variable 'onunhandledrejection' must be of type '(this: Window, ev: PromiseRejectionEvent) => any', but here has type '(this: DedicatedWorkerGlobalScope, ev: PromiseRejectionEvent) => any'.

node_modules/typescript/lib/lib.webworker.d.ts(5916,13): error TS2403: Subsequent variable declarations must have the same type.  Variable 'self' must be of type 'Window & typeof globalThis', but here has type 'WorkerGlobalScope & typeof globalThis'.
```

Reply:
Hmm, okay, seems fine.